src/utils/utils.cairo

```diff
@@ -590,4 +591,43 @@ namespace Helpers {
             data_len=data_len - 1, data=data + 1, bytes_len=bytes_len + 4, bytes=res
         );
     }
+
+    // @notice Initializes a dictionary of valid jump destinations in EVM bytecode.
+    // @param i The current index in the bytecode.
+    // @param bytecode_len The length of the bytecode.
+    // @param bytecode The EVM bytecode to analyze.
+    // @param valid_jumpdests A dictionary to be filled with valid jump destinations.
+    // @return valid_jumpdests The updated dictionary with valid jump destinations.
+    //
+    // @dev This function iterates over the bytecode from the current index 'i'.
+    // If the opcode at the current index is between 0x5f and 0x7f (PUSHN opcodes) (inclusive),
+    // it skips the next 'n_args' opcodes, where 'n_args' is the opcode minus 0x5f.
+    // If the opcode is 0x5b (JUMPDEST), it marks the current index as a valid jump destination.
+    // The function then increments the index and recursively calls itself until it has processed the entire bytecode.
+    func init_valid_jumpdests{range_check_ptr, valid_jumpdests: DictAccess*}(
```

Comment:
refactor this without tail recursion, and remove the `i` from the args. The signature should not get the destination dict as implicit arg, but the function should create the dict and return it.

This function could also be moved to accounts.cairo, there are already some `static` utils for accounts related functions